### PR TITLE
add a note on -Og and -O0 debug flags

### PR DESCRIPTION
**Pull request by davydden**
I tried to summarise the discussion and proposed solutions in https://github.com/dealii/dealii/issues/3737.

fixes https://github.com/dealii/dealii/issues/3737

### CODE REVIEW

File: doc/readme.html [hunk 1]
```diff
@@ -288,6 +288,15 @@
 	  are usually found almost instantaneously, displaying the file
 	  name and line number of where the problem occurred.
 	</p>
+  <p>
```
Comment (bangerth) [inline doc/readme.html]: Please indent to the same level as the text above.

File: doc/readme.html [hunk 2]
```diff
@@ -288,6 +288,15 @@
...
+    By default <code>-Og</code> compiler flag is used. It promises most of the
```
Comment (bangerth) [inline doc/readme.html]: -> Debug mode, by default, uses the ... flag

File: doc/readme.html [hunk 3]
```diff
@@ -288,6 +288,15 @@
...
+    debugging experience of <code>-O0</code> but at a moderate performance hit.
```
Comment (bangerth) [inline doc/readme.html]: But that's not right -- it promises *better* performance. Did you mean to reference `-O1` here?

File: doc/readme.html [hunk 4]
```diff
@@ -288,6 +288,15 @@
...
+    This is a reasonable choice for unit tests and trigger numerous asserts
```
Comment (bangerth) [inline doc/readme.html]: trigger -> enables

File: doc/readme.html [hunk 5]
```diff
@@ -288,6 +288,15 @@
...
+    within the library. Sometimes, however, one needs Debug mode to use
+    <code>-O0</code> (e.g. GCC 6.2.0 optimized out local variables with <code>-Og</code>).
```
Comment (bangerth) [inline doc/readme.html]: optimized -> optimizes

File: doc/readme.html [hunk 6]
```diff
@@ -288,6 +288,15 @@
 	  are usually found almost instantaneously, displaying the file
 	  name and line number of where the problem occurred.
 	</p>
+	<p>
+	  Debug mode, by default, uses the <code>-Og</code> flag. It promises most of the
```
Comment (kronbichler) [inline doc/readme.html]: `-Og` is a gcc-specific flag, on clang or Intel we actually use `-O0`.

File: doc/readme.html [hunk 7]
```diff
@@ -288,6 +288,15 @@
...
+	  debugging experience of <code>-O0</code> but at a better performance.
+	  This is a reasonable choice for unit tests and enables numerous asserts
+	  within the library. Sometimes, however, one needs Debug mode to use
+	  <code>-O0</code> (e.g. GCC 6.2.0 optimizes out local variables with <code>-Og</code>).
```
Comment (kronbichler) [inline doc/readme.html]: I guess not all users are familiar with what `-O0` actually means, so adding a description like "where all compiler optimizations are avoided and code and variables are exactly as indicated in the C++ program."

**Pull request merged**